crates/solvers/src/tests/legacy/mod.rs

```diff
@@ -48,18 +50,13 @@ pub async fn setup(expectations: Vec<Expectation>) -> SocketAddr {
 #[derive(Debug, Clone)]
 struct State(Arc<Mutex<Vec<Expectation>>>);
 
-/// Creates a temporary file containing the config of the given solver.
-pub fn create_temp_config_file(solver_addr: &SocketAddr) -> tempfile::TempPath {
-    let base_url = format!("http://{solver_addr}/solve");
-    let config = format!(
+/// Creates a legacy solver configuration for the specified host.
+pub fn config(solver_addr: &SocketAddr) -> tests::Config {
+    tests::Config::String(format!(
         r"
 solver-name = 'legacy_solver'
-solve-endpoint = '{}'
+solve-endpoint = 'http://{solver_addr}/solve'
```

Comment:
The balancer config calls this just `endpoint`. I don't really care about the specific name but I think we should make it consistent across config files if applicable.

Reply:
I saw you explicitly renamed `endpoint` to `solve-endpoint` (which makes sense for the legacy solver, since the endpoint is for `/solve`-ing). I'm not sure I like the name `solve-endpoint` here, as the Balancer SOR API is more of a DEXag-style API for quoting.

Reply:
I agree that `solve-endpoint` wouldn't fit well for the balancer solver but I think `endpoint` would be good enough for the `legacy` solver. There is only 1 endpoint that makes sense in that context so it's probably not needed to be overly specific. No strong opinion, though.